crypto/fipsmodule/sha/keccak1600.c

```diff
@@ -100,18 +55,56 @@ static const uint64_t iotas[] = {
     BIT_INTERLEAVE ? 0x8000808200000000ULL : 0x8000000080008008ULL
 };
 
-/*
- * This implementation is variant of KECCAK_1X above with outer-most
- * round loop unrolled twice. This allows to take temporary storage
- * out of round procedure and simplify references to it by alternating
- * it with actual data (see round loop below). Originally it was meant
- * rather as reference for an assembly implementation, but it seems to
- * play best with compilers [as well as provide best instruction per
- * processed byte ratio at minimal round unroll factor]...
- */
-static void Round(uint64_t R[5][5], uint64_t A[5][5], size_t i)
+#if !defined(KECCAK1600_ASM)
+
+#if defined(__i386) || defined(__i386__) || defined(_M_IX86) || \
+    (defined(__x86_64) && !defined(__BMI__)) || defined(_M_X64) || \
+    defined(__mips) || defined(__riscv) || defined(__s390__) || \
+    defined(__EMSCRIPTEN__)
+
+ // These platforms don't support "logical and with complement" instruction;
+# define KECCAK_COMPLEMENTING_TRANSFORM
+#endif
```

Comment:
Just curious: why bring this part of the code here instead of its original place above?

Reply:
I moved all the C function implementations under `#if !defined(KECCAK1600_ASM)` so that only `iotas` and  `rhotates` are kept in case of available assembly implementation.

Reply:
So `iotas` and  `rhotates` with the assembly implementation when you merge it in?